docs/dev/Lua API.rst

```diff
@@ -5766,82 +5766,129 @@ The names of the functions are also available as the keys of the
+  points on tiles which have the gear tile. ``gear_tiles`` is an optional array of two tiles that are
+  counted as gears in the workshop ascii tile raws.
+
+* ``setAnimationInfo(workshop_type, frames, frame_skip)``
+
+  Animate workshop by replacing displayed tiles (or graphical tiles). There are two ways this works:
+  if ``frame_skip>=0`` then it shows each frame for ``frame_skip`` of frames or if ``frame_skip<0``
+  Frames are synchronized to the machine this building is connected to.
+
+  :workshop_type:   custom workshop string id e.g. ``SOAPMAKER`` or numeric id
+  :frames:          table of frames. Each frame is sparse flat table with ids from ``0`` to ``31*31-1``.
+                    Each frame tile is table of integers from 4 to 8 members long. Tile members are as
+                    follow: tile, foreground color, background color, bright, graphics tile, overlay tile,
+                    signpost tile, item tile.
+  :frame_skip:      How many ticks to display one frame. If set to negative number (or skipped) frames
+                    are synchronized with machine animation.
+
+* ``setAnimationInfoAuto(name, make_graphics_too, frame_skip, gear_tiles)``
+
+  Animate workshop as with function above but generate frames automatically. This works by finding
+  tiles which have gears and animating them with alternating gear tiles.
+
+  :name:            custom workshop string id e.g. ``SOAPMAKER``
+  :make_graphics_too:  replace same tiles in graphics mode with tiles from vanilla df mechanism
+  :frame_skip:      How many ticks to display one frame. If set to negative number (or skipped) frames
+                    are synchronized with machine animation.
+  :gear_tiles:      Optional array of 2 or 4 indexes. First two define ascii tiles and next two graphics tiles
+
+* ``setOnUpdate(name,interval,callback)``
+
+  Setup callback to be called every ``interval`` of ticks for each building of this type. Note: low interval
+  numbers and/or many workshops that use this might reduce DF performance.
+
+  :name:   custom workshop string id e.g. ``SOAPMAKER``
+  :interval:   how many ticks to skip between event triggers
+  :callback:   function to call
+
+* ``getPower(building)``
+
+  Returns two number - produced and consumed power if building can be modified and returns nothing otherwise.
+
+  :building:   specific workshop that produces or consumes power
+
+* ``setPower(building,power_consumed, power_produced)``
+
+  Sets current power production and consumption for a building. Can be used to make buildings that
+  dynamically change power consumption and production.
+
+  :building:   specific workshop that produces or consumes power
+  :power_consumed:   set building to consume this amount of power
+  :power_produced:   output this amount of power
+
+
+Events
+------
+
+This module exports two events. However only one is documented here and is intended to be used directly. To use
+``onUpdateAction`` instead call ``setOnUpdate`` function.
```

Comment:
why does onUpdateAction exist as a Lua event, then?

Reply:
The plugin lua part still uses that event (i.e. `setOnUpdate` function)

Reply:
I'm not convinced this is sufficient reasoning. Why do you need a lua event to communicate with your own Lua layer? Why can't you call the Lua functions directly?